src/core/config/ip6.h

```diff
@@ -177,4 +177,34 @@
 #define OPENTHREAD_CONFIG_UNSECURE_TRAFFIC_MANAGED_BY_STACK_ENABLE 0
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_MAX_MULTICAST_LISTENER_NUM
+ *
+ * The maximum number of supported IPv6 multicast addresses (with scope >= 4) allows
+ * to be externally added by the device ifself or registered by its child.
+ * At least should be the same as OPENTHREAD_CONFIG_IP6_MAX_EXT_MCAST_ADDRS for MTD device.
+ * And allocate more for FTD in case it would proxy additional Multicast address its MTD children
+ * registered.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_MAX_MULTICAST_LISTENER_NUM
+#if OPENTHREAD_FTD
+#define OPENTHREAD_CONFIG_MAX_MULTICAST_LISTENER_NUM (OPENTHREAD_CONFIG_IP6_MAX_EXT_MCAST_ADDRS * 3)
+#elif OPENTHREAD_MTD
+#define OPENTHREAD_CONFIG_MAX_MULTICAST_LISTENER_NUM (OPENTHREAD_CONFIG_IP6_MAX_EXT_MCAST_ADDRS)
+#endif
+#endif
+
+/*
+ * For certification test purpose, force `OPENTHREAD_CONFIG_MAX_MULTICAST_LISTENER_NUM` to be
```

Comment:
The macro was introduced for certification purpose to subscribe/register to a bunch of multiple address once by cli command which was mainly designed for THCI
https://github.com/openthread/openthread-1.2/blob/f1c71c2236a39e8491a9f8cb7819b521c8495fd5/src/cli/cli.cpp#L1628-L1633

I think newer THCI should not require such behavior any more. 

Thoughts?

Reply:
I will verify